File: CS/DXSample/CustomRemoveColumnDropTarget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpf.Grid.Native;
using System.Windows;
using DevExpress.Xpf.Core;
using DevExpress.Xpf.Grid;

namespace DXSample
{
    public class CustomRemoveColumnDropTarget : RemoveColumnDropTarget
    {
        public override void Drop(UIElement source, Point pt)
        {
            TableView view = GetDataView(source) as TableView;
            GridColumn column = (source as GridColumnHeader).DataContext as GridColumn;
            bool flag = Convert.ToBoolean(column.GetValue(AbsoluteVisible.IsEnableProperty));
            if (flag)
            {
                DragDropScroller.StopScrolling(GetDataView(source));
                return;
            }
            base.Drop(source, pt);
        }
    
    }
}

File: VB/DXSample/CustomRemoveColumnDropTarget.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Xpf.Grid.Native
Imports System.Windows
Imports DevExpress.Xpf.Core
Imports DevExpress.Xpf.Grid

Namespace DXSample
    Public Class CustomRemoveColumnDropTarget
        Inherits RemoveColumnDropTarget

        Public Overrides Sub Drop(ByVal source As UIElement, ByVal pt As Point)
            Dim view As TableView = TryCast(GetDataView(source), TableView)
            Dim column As GridColumn = TryCast((TryCast(source, GridColumnHeader)).DataContext, GridColumn)
            Dim flag As Boolean = Convert.ToBoolean(column.GetValue(AbsoluteVisible.IsEnableProperty))
            If flag Then
                DragDropScroller.StopScrolling(GetDataView(source))
                Return
            End If
            MyBase.Drop(source, pt)
        End Sub

    End Class
End Namespace
